playground/compiler/tsconfig.json

```diff
@@ -25,6 +26,8 @@
     "paths": {
       "~/*": ["./app/*"]
     },
-    "noEmit": true
+    "noEmit": true,
+    "plugins": [{ "name": "@react-router/dev" }],
```

Comment:
It's nice that we show how to use the plugin in the playground apps, but we should go one step further in this PR and show how those types are actually used in these apps.

Reply:
Added some example usages in both routes of `playground/compiler`